extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java

```diff
@@ -299,17 +300,29 @@ private void warnIfJaxRsPathUsed(IndexView index, DotName healthAnnotation) {
     }
 
     @BuildStep
-    public void kubernetes(NonApplicationRootPathBuildItem frameworkRootPath,
+    public void kubernetes(HttpBuildTimeConfig httpConfig, NonApplicationRootPathBuildItem frameworkRootPath,
             SmallRyeHealthConfig healthConfig,
             BuildProducer<KubernetesHealthLivenessPathBuildItem> livenessPathItemProducer,
             BuildProducer<KubernetesHealthReadinessPathBuildItem> readinessPathItemProducer) {
 
         livenessPathItemProducer.produce(
                 new KubernetesHealthLivenessPathBuildItem(
-                        frameworkRootPath.adjustPath(healthConfig.rootPath + healthConfig.livenessPath)));
+                        kubernetesPath(httpConfig, frameworkRootPath, healthConfig.rootPath, healthConfig.livenessPath)));
         readinessPathItemProducer.produce(
                 new KubernetesHealthReadinessPathBuildItem(
-                        frameworkRootPath.adjustPath(healthConfig.rootPath + healthConfig.readinessPath)));
+                        kubernetesPath(httpConfig, frameworkRootPath, healthConfig.rootPath, healthConfig.readinessPath)));
+    }
+
+    private String kubernetesPath(HttpBuildTimeConfig httpConfig, NonApplicationRootPathBuildItem frameworkRootPath,
+            String healthRootPath, String specificPath) {
+        String result = frameworkRootPath.adjustPath(healthRootPath + specificPath);
+        if (!httpConfig.rootPath.equals("/")) {
+            String prefix = httpConfig.rootPath.endsWith("/")
```

Comment:
Can `httpConfig.adjustPath()` be used instead of the manual path manipulation?

Reply:
Good idea. I'll add that method as it currently doesn't exist

Reply:
PR updated as per your suggestion

Reply:
It doesn't? Thought it did as I'd done the same for non app endpoint.

Reply:
Maybe it was in some other http related config